src/store/global/global.thunks.ts

```diff
@@ -440,6 +441,68 @@ function checkPendingDelayedWithdrawals(): AppThunk {
   };
 }
 
+/**
+ * Checks if we have some exit that is not saved in localStorage
+ */
+function recoverPendingDelayedWithdrawals(exits: Exits): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, pendingDelayedWithdraws, ethereumNetworkTask },
+    } = getState();
+    if (wallet !== undefined && ethereumNetworkTask.status === "successful") {
+      const {
+        data: { chainId },
+      } = ethereumNetworkTask;
+      if (chainId !== undefined) {
+        const { hermezEthereumAddress } = wallet;
+        const provider = Providers.getProvider();
+        const {
+          data: { chainId },
+        } = ethereumNetworkTask;
+        const accountPendingDelayedWithdraws: PendingDelayedWithdraw[] =
+          storage.getPendingDelayedWithdrawsByHermezAddress(
+            pendingDelayedWithdraws,
+            chainId,
+            hermezEthereumAddress
+          );
+        const batchNumPendingDelayedWithdraws = accountPendingDelayedWithdraws.map(
+          (account) => account.batchNum
+        );
+
+        exits.exits.map(async (exit) => {
+          if (
+            !batchNumPendingDelayedWithdraws.includes(exit.batchNum) &&
+            exit.delayedWithdrawRequest
+          ) {
+            try {
```

Comment:
`getBlockWithTransactions` could fail requesting the transactions

Reply:
It feels strange to me using `async` and then having to wrap the code in a `try/catch`. A promise already provides a `try/catch` around it. If an exception happens in `getBlockWithTransactions` it would be caught and treated as a rejection.

It's probably a style issue...

Reply:
Merging this comment with the other one about the `map` vs `forEach` this looks simpler and more correct to me:

```typescript
exits.exits.forEach((exit) => {
  if (
    !batchNumPendingDelayedWithdraws.includes(exit.batchNum) &&
    exit.delayedWithdrawRequest
  ) {
    void provider
      .getBlockWithTransactions(exit.delayedWithdrawRequest)
      .then((blockWithTransactions) => {
        const pendingDelayedWithdraw = blockWithTransactions.transactions.find(
          (tx) => Addresses.getEthereumAddress(hermezEthereumAddress) === tx.from
        );
        if (pendingDelayedWithdraw) {
          dispatch(
            addPendingDelayedWithdraw({
              ...exit,
              hash: pendingDelayedWithdraw.hash,
              id: `${exit.accountIndex}${exit.batchNum}`,
              hermezEthereumAddress: wallet.hermezEthereumAddress,
              isInstant: false, // TODO I'll remove this key that it's unused with the ExitCard refactor
              timestamp: new Date(blockWithTransactions.timestamp * 1000).toISOString(),
            })
          );
        }
      });
  }
});
```

* No map, since we're not transforming any value
* No try/catch since the promise already provides it
* No need for `async/await`

Does it make sense to you?